### DISCUSSION TITLE
Add configuration option for postgres port

### DISCUSSION THREAD

**Discussion by davidmehren** [nextcloud/all-in-one#3097] [Ideas] [answered]
The Postgres port `5432` is hardcoded at various places.
This makes it impossible to use an external Postgres database with a non-default port.

I propose to add a variable `POSTGRES_PORT` that can be used to set the database port.

**Comment by szaimen** [comment]
Can you explain your exact use case? 

Do you want to use the manual-install method and use an external postgres database in there? 

Also why arent you able to use the default port?

**Reply by lekhnath** [reply]
What do you mean by exact use case?  There are million of use case where we want to provide custom port number for postgres connection. This is fairly simple configuration option to have. I am running multiple postgres clusters in a single server. Each cluster must have different port number. I've enabled those specific ports in firewall setting. But I did not see any option to provide port number in next cloud configuration. This was unexpected for me.

**Comment by thegcat** [comment]
> Can you explain your exact use case?

We use a centralised multi-replica fail-over PG cluster for all services we host. This ensures HA on the PG side, less admin overhead than one for each service, and better resource usage than one for each service. Failover connections and distribution to the current primary are handled via a pgbouncer, which in our installation listens on another port than the default pg port.

> Do you want to use the manual-install method and use an external postgres database in there?

See above, though I wasn't directly involved in the whole planning, my understanding is that we plan on using AIO with our own Postgres. If that is only supported on manual-install that will probably be what is planned, yes.

> Also why arent you able to use the default port?

The short answer is because it's configured this way and changing it would mean rearchitecting parts of our setup. We understand adding a configuration option is also devoid of work, we'd be willing to have a crack at a PR if there's interest in this feature.

(I also had a little longer and snarkier answer, it would probably not bring things forward, so not for today)

**Reply by davidmehren** [reply]
@szaimen Would you be interested in us contributing an option for configuring the PG port?

**Reply by szaimen** [reply]
If you still need this then yes

**Comment by nooblag** [comment]
Hardcoding the port is a bad idea, and also ignoring the conf.php settings is a problem: https://github.com/nextcloud/server/issues/45496

**Reply by szaimen** [reply, marked as answer]
The dbport can be adjusted since a while now: https://github.com/nextcloud/all-in-one/pull/4433

**Reply by nooblag** [reply]
Sorry, I posted this in the wrong place. My issue is for `server` not `all-in-one`. Apologies.

**Reply by szaimen** [reply]
All right. No problem
